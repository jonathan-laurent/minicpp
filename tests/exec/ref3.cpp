#include <iostream>

void f(int &y) {
  y = y + 1;
}

int main() {
  int x = 41;
  std::cout << "x = " << x << "\n";
  f(x);
  std::cout << "x = " << x << "\n";
  f(x);
  std::cout << "x = " << x << "\n";
  x = 0;
  std::cout << "x = " << x << "\n";
  f(x);
  std::cout << "x = " << x << "\n";
}
