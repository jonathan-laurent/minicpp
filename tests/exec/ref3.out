x = 41
x = 42
x = 43
x = 0
x = 1
